components/server/ee/src/gitlab/gitlab-app-support.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+ * Licensed under the Gitpod Enterprise Source Code License,
+ * See License.enterprise.txt in the project root folder.
+ */
+
+import { ProviderRepository, User } from "@gitpod/gitpod-protocol";
+import { inject, injectable } from "inversify";
+import { TokenProvider } from "../../../src/user/token-provider";
+import { UserDB } from "@gitpod/gitpod-db/lib";
+import { Gitlab } from "@gitbeaker/node";
+
+@injectable()
+export class GitLabAppSupport {
+
+    @inject(UserDB) protected readonly userDB: UserDB;
+    @inject(TokenProvider) protected readonly tokenProvider: TokenProvider;
+
+    async getProviderRepositoriesForUser(params: { user: User, provider: string, hints?: object }): Promise<ProviderRepository[]> {
```

Comment:
**issue**: Tried on purpose logging in with GitHub only so that no there’s no active integration with GitLab and got the following error in console:

```js
Error: Request getProviderRepositoriesForUser failed with message: User xyz has no identity for host: gitlab.com!

Reply:
Nice! Thanks! `UserContext` isn't updated properly.